From 711f4993e19160f0d8eaf75b2b9afaab7e3e5d8e Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Mon, 9 Jul 2018 16:47:50 +0530
Subject: [PATCH 4/6] Fix compilation error when BTM_SSR_INCLUDED is set to
 false

issue :
Sniff subrating is disabled by setting the BTM_SSR_INCLUDED to false.
This causes compilation errors in stack.

fix :
This patch address the compilation failures

Change-Id: Id18942e6a144db310079d5a5e990d337c01bd96c
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-65440
Signed-off-by: anitha3x <anithax.h.chandrasekar@intel.com
---
 bta/dm/bta_dm_pm.cc | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/bta/dm/bta_dm_pm.cc b/bta/dm/bta_dm_pm.cc
index 23716c1..f41af0b 100644
--- a/bta/dm/bta_dm_pm.cc
+++ b/bta/dm/bta_dm_pm.cc
@@ -47,7 +47,6 @@ static void bta_dm_pm_btm_cback(const RawAddress& bd_addr,
 static bool bta_dm_pm_park(const RawAddress& peer_addr);
 static bool bta_dm_pm_sniff(tBTA_DM_PEER_DEVICE* p_peer_dev, uint8_t index);
 static bool bta_dm_pm_is_sco_active();
-static int bta_dm_get_sco_index();
 static void bta_dm_pm_hid_check(bool bScoActive);
 static void bta_dm_pm_set_sniff_policy(tBTA_DM_PEER_DEVICE* p_dev,
                                        bool bDisable);
@@ -61,6 +60,7 @@ static void bta_dm_pm_stop_timer_by_index(tBTA_PM_TIMER* p_timer,
  * can use it */
 #define BTA_DM_PM_SSR_HH BTA_DM_PM_SSR1
 #endif
+static int bta_dm_get_sco_index();
 static void bta_dm_pm_ssr(const RawAddress& peer_addr);
 #endif
 
@@ -331,11 +331,11 @@ static void bta_dm_pm_stop_timer_by_index(tBTA_PM_TIMER* p_timer,
 static void bta_dm_pm_cback(tBTA_SYS_CONN_STATUS status, uint8_t id,
                             uint8_t app_id, const RawAddress& peer_addr) {
   uint8_t i, j;
-  uint8_t* p = NULL;
   tBTA_DM_PEER_DEVICE* p_dev;
   tBTA_DM_PM_REQ pm_req = BTA_DM_PM_NEW_REQ;
 
 #if (BTM_SSR_INCLUDED == TRUE)
+  uint8_t* p = NULL;
   int index = BTA_DM_PM_SSR0;
 #endif
 
@@ -696,8 +696,8 @@ static bool bta_dm_pm_sniff(tBTA_DM_PEER_DEVICE* p_peer_dev, uint8_t index) {
 #endif
 
   BTM_ReadPowerMode(p_peer_dev->peer_bdaddr, &mode);
-  p_rem_feat = BTM_ReadRemoteFeatures(p_peer_dev->peer_bdaddr);
 #if (BTM_SSR_INCLUDED == TRUE)
+  p_rem_feat = BTM_ReadRemoteFeatures(p_peer_dev->peer_bdaddr);
   APPL_TRACE_DEBUG("bta_dm_pm_sniff cur:%d, idx:%d, info:x%x", mode, index,
                    p_peer_dev->info);
   if (mode != BTM_PM_MD_SNIFF ||
@@ -1074,7 +1074,7 @@ static bool bta_dm_pm_is_sco_active() {
   APPL_TRACE_DEBUG("bta_dm_is_sco_active: SCO active: %d", bScoActive);
   return bScoActive;
 }
-
+#if (BTM_SSR_INCLUDED == TRUE)
 /*******************************************************************************
  *
  * Function        bta_dm_get_sco_index
@@ -1094,7 +1094,7 @@ static int bta_dm_get_sco_index() {
   }
   return -1;
 }
-
+#endif // BTM_SSR_INCLUDED
 /*******************************************************************************
  *
  * Function         bta_dm_pm_hid_check
-- 
1.9.1

